DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -0,0 +1,83 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Rules;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class AnonymizerEngine
+    {
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerEngine>();
+        private readonly AnonymizerSettings _anonymizerSettings;
+        private readonly AnonymizerConfigurationManager _configurationManager;
+        private readonly IAnonymizerRuleFactory _ruleFactory;
+
+        public AnonymizerEngine(string configFilePath = "configuration.json", AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+            : this(AnonymizerConfigurationManager.CreateFromConfigurationFile(configFilePath), anonymizerSettings, ruleFactory)
+        {
+        }
+
+        public AnonymizerEngine(AnonymizerConfigurationManager configurationManager, AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+        {
+            EnsureArg.IsNotNull(configurationManager, nameof(configurationManager));
+
+            _anonymizerSettings = anonymizerSettings ?? new AnonymizerSettings();
+            _configurationManager = configurationManager;
+            _ruleFactory = ruleFactory ?? new AnonymizerRuleFactory(_configurationManager.Configuration, new DicomProcessorFactory(), new AnonymizerSettingsFactory());
+            _logger.LogDebug("Successfully initialized anonymizer engine.");
+        }
+
+        public void AnonymizeDataset(DicomDataset dataset)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+
+            var context = InitContext(dataset);
+            ValidateInput(dataset);
+            dataset.AutoValidate = false;
```

Comment:
dataset.AutoValidate is default be true in fo-dicom library. And the library will auto check the value when insert an item.
Since we add an option : ValidateOutput. we need to close this autoValidate.

Reply:
after you validate, do you need to restore this value?

Reply:
the dataset will be discarded after anonymization.

Reply:
Will be saved to file after anonymization.